### ISSUE TITLE
Proposal: store info to verify the chain against a validator quorum.

### ISSUE THREAD

**Issue by sstanfield** [Telcoin-Association/telcoin-network#88]
Prerequisite: https://github.com/Telcoin-Association/telcoin-network/issues/87

We need to be able to not only verify the chain is correct (i.e. that all blocks are well formed) but also that block(s) are valid in the sense that a quorum of validators signed off on them.

We might be able to handle it this way:
- When a series of blocks are committed to the chain via consensus (see payload_builder.rs/execute_consenses_output() for example) save the WorkerBlock and Certificates for the leaders block.  This would save one WorkerBlock/certs per commit not one for each block (just the leader block not all the blocks it has as parents).
- When we want to verify a block is "authentic" we would need to look ahead until we find a block that contains a saved WorkerBlock/certs (should not have to go far, we might also be able to store this block in block info as well).
- Once we have a future block then check the WorkerBlock against its certificates and make sure there is a quorum.  Note this step will need access to the committee that was in effect at it's block height, probably use a smart contract for this.
- If the worker block is valid then replay it against the parent to make sure the actual block is correct.
- Once it is validated then "walk back" the chain one parent at a time and verify those blocks are well formed.  Since we verified a future block then logically any parent block would also be considered valid (else it would not have been a valid parent).  Repeat until we verify the block we wanted to verify.

This process could be repeated to genesis although it would make sense to reverify blocks with WorkingBlock/certs as we go.  The process could also be used "forward" by verifying any blocks with WorkerBlocks saved and making sure other blocks were well formed.

Note: if a WorkerBlock verification fails then all blocks back to the last verified WorkerBlock have to be considered invalid with this scheme.

This would:
- Allow the chain/blocks to be verified as not only well formed but also as selected by validators (i.e. canonical).
- Should not use to much storage because we are only saving one WorkerBlock/certs per consensus every other round which leave a lot of blocks without this extra data.
- Trades some speed for this trimmer data usage.
- The chain itself would contain all the data needed to validate itself without re-playlng consensus.

Note that tracking consensus and being able to replay that are a separate issue and also need to be supported.  This algorithm is about what can we store unchain in order to verify the chains integrity when needed.

**Comment by sstanfield** [comment]
I think the consensus chain makes this OBE.
